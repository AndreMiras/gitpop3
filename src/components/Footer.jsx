import React from "react";
import { Container } from "react-bootstrap";
import { version } from "../../package.json";

const Footer = () => (
  <footer className="footer d-none d-md-block">
    <Container className="text-center">
      <span>Copyright &copy; Andre Miras 2020 - gitpop3 v{version}</span>
    </Container>
  </footer>
);

export default Footer;
