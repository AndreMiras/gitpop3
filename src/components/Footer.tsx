import { FunctionComponent } from "react";
import { Container } from "react-bootstrap";
import { version } from "../../package.json";

const Footer: FunctionComponent = () => (
  <footer className="footer d-none d-md-block">
    <Container className="text-center">
      <span>Copyright &copy; Andre Miras 2020 - gitpop3 v{version}</span>
    </Container>
  </footer>
);

export default Footer;
