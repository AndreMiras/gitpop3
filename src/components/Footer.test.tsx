import { describe, it, test, expect, vi } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import Footer from "./Footer";

test("renders", () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("displays version from package.json", () => {
  const { version } = require("../../package.json");
  const tree = renderer.create(<Footer />).toJSON();
  expect(JSON.stringify(tree)).toContain(version);
});
