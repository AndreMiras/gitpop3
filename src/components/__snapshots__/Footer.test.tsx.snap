// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders 1`] = `
<footer
  className="footer d-none d-md-block"
>
  <div
    className="text-center container"
  >
    <span>
      Copyright Â© Andre Miras 2020 - gitpop3 v
      2020.12.19
    </span>
  </div>
</footer>
`;
